siva/library.go

```diff
@@ -90,24 +88,23 @@ func NewLibrary(
 		ops = &(*options)
 	}
 
-	metadata, err := loadLibraryMetadata(fs)
-	if err != nil {
-		// TODO: skip metadata if corrupted?
-		return nil, err
-	}
+	var (
+		metadata *libMetadata
+		err      error
+	)
 
-	if metadata == nil {
-		metadata = NewLibraryMetadata("", -1)
+	if ops.MetadataReadOnly {
+		metadata, err = loadLibraryMetadata(fs)
+	} else {
+		metadata, err = loadOrCreateLibraryMetadata(id, fs)
 	}
 
-	if id == "" {
-		if metadata.ID == "" && !ops.DontGenerateID {
-			metadata.GenerateID()
-		}
+	if err != nil && !os.IsNotExist(err) {
+		return nil, err
+	}
 
+	if metadata != nil {
```

Comment:
Shouldn't be `if metadata != nil && id == "" {`? There's the case where you provide an ID but there's metadata. Shouldn't we use the provided p `id`?

Reply:
If that is the default case we want, yes it has to be changed.

so the default would be overwrite the id in case the given one differs from the metadata id, what would we do with `MetadataReadOnly` in that case?

Reply:
In that case there's no write unless a save operation is explicitly called and I believe `SaveMetadata` was deleted.